polyfills/Window.location.origin/detect.js

```diff
@@ -0,0 +1,10 @@
+(function(global) {
+	'use strict';
+
+	try {
+		if (global.window) {
+			return window.location.origin
+		}
+	}
+	catch (error) { }
+}(this))
```

Comment:
I think a way of expressing this that's more consistent with our existing detects would be:

```javascript
'origin' in window.location
```

If this isn't good enough, wouldn't lots of our other detects be flawed for the same reason?  We should potentially have something in the contributors' guide about a standard way of testing for the presence of an object property.